### PR TITLE
Fixed user age calculator on userhover

### PR DESCRIPTION
**Pull request by yangzihe**
I fixed the bug showing wrong age of certain accounts. Also added a little bit of documentation to explain logic for new code.

</img>

### CODE REVIEW

**Comment by allthefoxes** [comment]
Magic words:

Fixes #2058

File: lib/core/utils.js [hunk 1]
```diff
@@ -1105,9 +1105,9 @@ RESUtils.niceDateDiff = function(origdate, newdate) {
 	var aday = origdate.getUTCDate();
 	var ayear = origdate.getUTCFullYear();
 
-	var tyear = newdate.getUTCFullYear();
 	var tmonth = newdate.getUTCMonth() + 1;
 	var tday = newdate.getUTCDate();
+	var tyear = newdate.getUTCFullYear();
```
Comment (andytuba) [inline lib/core/utils.js]: tiny style note: why did you move this?

**Comment by andytuba** [comment]
Yay, thanks for math fixes!

File: lib/core/utils.js (hunk 1 above)
Comment (yangzihe) [inline lib/core/utils.js]: Oh because I wanted to match the order of the original date variables. I guess it's a little bit nitpicky, but I saw no reason why the variables wouldn't have the same order in the first place, so I just shifted the line.

Comment (honestbleeps) [inline lib/core/utils.js]: I like the cut of your jib, sir.

Comment (andytuba) [inline lib/core/utils.js]: fair nuff.
